src/librustc_middle/middle/stability.rs

```diff
@@ -280,9 +280,15 @@ impl<'tcx> TyCtxt<'tcx> {
     /// If `id` is `Some(_)`, this function will also check if the item at `def_id` has been
     /// deprecated. If the item is indeed deprecated, we will emit a deprecation lint attached to
     /// `id`.
-    pub fn eval_stability(self, def_id: DefId, id: Option<HirId>, span: Span) -> EvalResult {
+    pub fn eval_stability(
+        self,
+        def_id: DefId,
+        id: Option<HirId>,
+        span: Span,
+        check_deprecation: bool,
```

Comment:
Why do we not always want to check deprecation?

Reply:
The answer was https://github.com/rust-lang/rust/pull/65083#discussion_r368759062, but now that `init<T>()` is gone it can be removed.

It may have to be added again if a similar situation arises.
However, if a new redundant deprecation message arises it won't break additional existing code, like this PR would have before `init`'s removal.

If I recall the interaction involved deprecated parents of generic parameters, and sub-crates or possible intrinsics.
I tried to add some tests, but they didn't trigger the previous issue.

```rust
#[rustc_deprecated(since = "1.1.0", reason = "test")]
#[stable(feature = "stable_test_feature", since = "1.0.0")]
pub struct Struct6<A = usize> {
    field: A,
}

#[rustc_deprecated(since = "1.1.0", reason = "test")]
#[stable(feature = "stable_test_feature", since = "1.0.0")]
pub struct Struct7<A> {
    field: A,
}

#[stable(feature = "stable_test_feature", since = "1.0.0")]
pub const STRUCT6: Struct6 = Struct6 { field: 1 };

#[stable(feature = "stable_test_feature", since = "1.0.0")]
pub const STRUCT7: Struct7<usize> = Struct7 { field: 1 };
```

Reply:
Okay, let's remove this special casing for now, and we can revisit it if it occurs again. Unless we have a concrete test case demonstrating it, I'd prefer simpler behaviour.